gcovr/formats/html/write.py

```diff
@@ -69,15 +68,25 @@ def __call__(self, *args):
         return self.get(*args)
 
 
-# Loading Jinja and preparing the environmen is fairly costly.
+# Loading Jinja and preparing the environment is fairly costly.
 # Only do this work if templates are actually used.
 # This speeds up text and XML output.
 @Lazy
-def templates():
-    from jinja2 import Environment, PackageLoader
+def templates(options):
+    from jinja2 import ChoiceLoader, Environment, FileSystemLoader, PackageLoader
+
+    if options.html_template_dir is None:
+        loader = PackageLoader("gcovr.formats.html")
+    else:
+        loader = ChoiceLoader(
```

Comment:
I don't understand the test failure in the CI. Even if I put a exception here it isn't raised with the injection class and I don't know why.

I suggest to remove this test since you've also added a makefile test.